### DISCUSSION TITLE
Timeout during first invocation of serverless function—slow boot?

### DISCUSSION THREAD

**Discussion by henningko** [vercel/community#334] [Help]
Cheers,

I am deploying a Nuxt3 app and using Nuxt's API. I have an endpoint that parses URLs and extracts texts using [mozilla/readability](https://github.com/mozilla/readability), and then saves it in supabase:

`/api/add.ts`
```ts
import type { IncomingMessage, ServerResponse } from "http";
import supabase from "../lib/supabase";
import { useBody } from "h3";
import { JSDOM } from "jsdom";
import { Readability } from "@mozilla/readability";
import sanitizeHtml from "sanitize-html";

const allowedTags = [...];
export default async (req: IncomingMessage, res: ServerResponse) => {
  const { url, access_token } = await useBody(req);
  const response = await $fetch(url);
  const { origin } = new URL(url);
  const doc = new JSDOM(response, { url: origin });
  const reader = new Readability(doc.window.document);
  const article = reader.parse();
  const clean = sanitizeHtml(article.content, { allowedTags: allowedTags });
  let session = supabase.auth.session();
  if (!session.user?.id) {
    session = supabase.auth.setAuth(access_token);
    ({ user: session.user } = await supabase.auth.api.getUser(access_token));
  }
  if (session.user?.id) {
    try {
      const { data, error } = await supabase.from("user_content").insert([
        {
          title: article.title,
          data: clean,
          type: "article",
          user_id: session.user.id,
          format: "text/html",
          target_url: url,
        },
      ]);
      return { success: "Added successfully" };
    } catch (error) {
      res.statusCode = 500;
      return error;
    }
  } else {
    res.statusCode = 401;
    return "Unauthorized";
  }
};
```
Full codebase: https://github.com/henningko/briefxyz

This works great as long as the function is "warm"—the request runs in well under 2sec. However, during first invocation, the request routinely times out with `504 FUNCTION_INVOCATION_TIMEOUT`.

I'm in the free tier, which puts the timeout at 5sec, but this seems to be a question of a slow boot rather than a slow script (albeit, there's probably room for improvement).

Is there any way to speed up the response on first invocation?

**Comment by samcx** [comment]
@henningko There is no way to speed up the response on first invocation — all Serverless Functions execute equally.

Have you tried using a different region to see if it would finish executing in a shorter period of time? →

![CleanShot 2022-02-12 at 01 30 23@2x](https://user-images.githubusercontent.com/28912696/153705800-b16608de-b533-47bd-9416-e32696e7994c.jpg)
